lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -258,15 +272,21 @@ def single_thread(request, course_key, discussion_id, thread_id):
     """
     Renders a response to display a single discussion thread.
     """
-
     nr_transaction = newrelic.agent.current_transaction()
 
     course = get_course_with_access(request.user, 'load_forum', course_key)
-    course_settings = make_course_settings(course)
+    course_settings = make_course_settings(course, request.user)
     cc_user = cc.User.from_django_user(request.user)
     user_info = cc_user.to_dict()
     is_moderator = cached_has_permission(request.user, "see_all_cohorts", course_key)
 
+    # Verify that the student has access to this thread if belongs to a discussion module
+    accessible_discussion_ids = [
+        module.discussion_id for module in utils.get_accessible_discussion_modules(course, request.user)
+    ]
+    if discussion_id not in set(course.top_level_discussion_topic_ids + accessible_discussion_ids):
```

Comment:
this construction is what i had in mind with my comment above on L135.  seeing that it's used in multiple places i'd suggest creating a function in `utils` that takes a course and a user and gives back the list (or set) of ids